src/itemlistformaction.cpp

```diff
@@ -1496,17 +1496,13 @@ void ItemListFormAction::handle_op_saveall()
 		return;
 	}
 
-	std::string directory = v->run_dirbrowser();
+	Filepath directory = v->run_dirbrowser();
 
-	if (directory.empty()) {
+	if (directory.to_locale_string().empty()) {
 		return;
 	}
```

Comment:
Empty path is a kludge, let's get rid of it :)

1. on `master`, make `run_dirbrowser()` return `nonstd::optional<std::string>`;
2. wait for that change to get merged;
3. rebase this branch onto `master` and while doing so, update the code here to check if the `optional` is empty.

Reply:
https://github.com/newsboat/newsboat/pull/2593